R/processDataNew.R

```diff
@@ -97,7 +100,107 @@ fixMatrixO <- function(O){
   O
 }
 
-processData <- function(inputs){
+#' Infer column names for Input O to "variable", "coefficient", "lb", "ub", "type"
+#'
+#' @param O A data.frame, original O matrix from input Anchor O.
+#' @param displayFlag A boolean, if display field map for Input O in the UI.
+inferO <- function(O, config) {
+  displayFlag <- config$displayFieldMapO
+  r <- c("variable", "coefficient", "lb", "ub", "type")
+  names(r) <- config[c("nameVar", "nameCoef", "nameLower", "nameUpper", "nameType")]
+  if (displayFlag) {
+    O <- plyr::rename(O, r)
+  }
+  return(O)
+}
+
+#' Infer column names for Input A based on constraintMode
+#'
+#' @param A A data.frame, original A matrix from input Anchor A
+#' @param constrMode A string, constraint mode.
+inferA <- function(A, constrMode) {
+  if (constrMode == 'conInRow') {
+    constraint <- names(Filter(function(x){return(!isTRUE(x))},
+                               sapply(A, is.numeric)))
+    if (is.null(constraint) || length(constraint) == 0) {
```

Comment:
I would refactor this into its own function

```
findStringFields <- function(x){
  names(Filter(isTRUE, lapply(x, Negate(is.numeric))))
}
```

This will make the intent clearer and keep the code DRY.